### PR TITLE
wikicurses: init at 1.3

### PR DESCRIPTION
**Pull request by matthiasbeyer**
**WIP** - suggestions welcome
###### Things done
- [ ] Tested using sandboxing
  ([nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS,
    or option `build-use-chroot` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [ ] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/wikicurses/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{ stdenv, fetchurl, buildPythonApplication, python3, pythonPackages }:
+
```
Comment (FRidh) [inline pkgs/applications/misc/wikicurses/default.nix]: you can remove python3 and buildPythonApplication as they're attributes of pythonPackages

Comment (FRidh) [inline pkgs/applications/misc/wikicurses/default.nix]: Furthermore, you're now mixing python3 interpreter with python2 packages.

**Comment by matthiasbeyer** [comment]
@FRidh does the latest commit fix the issues you describe?

Either way, I'm still getting

```
error: value is a string while a set was expected
```

and I don't know why.

**Comment by FRidh** [comment]
I don't know why you would get that error. It fails here with
```
$ nix-shell -p wikicurses -I nixpkgs=.
error: beautifulsoup-3.2.1 not supported for interpreter python3.5m
```
You need to use `beautifulsoup4` instead of `beautifulsoup`

**Comment by matthiasbeyer** [comment]
I use `nix-build -A wikicurses` here and it fails.

I fixed the issue with `beautifulsoup4`.

**Comment by FRidh** [comment]
callPackages -> callPackage

File: pkgs/applications/misc/wikicurses/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    url = "http://github.com/ids1024/wikicurses/archive/v${version}.tar.gz";
+    sha256 = "1yxgafk1sczg1xi2p6nhrvr3hchp7ydw98n48lp3qzwnryn1kxv8";
+  };
+
+  propagatedBuildInputs = with pythonPackages; [ urwid beautifulsoup4 lxml ];
+
+  meta = {
+    description = "A simple curses interface for MediaWiki sites such as Wikipedia";
+    homepage = "https://github.com/ids1024/wikicurses/";
+    license = stdenv.lib.licenses.mit; # not sure here
```
Comment (fpletz) [inline pkgs/applications/misc/wikicurses/default.nix]: You can remove that comment. The license is indeed MIT as stated in setup.py.

**Comment by matthiasbeyer** [comment]
Nice that github detects that you squashed these commits! Thank you, @fpletz, for merging.

**Pull request merged**